string type
int16 tunnel
int16 row
int16[] edge_nodes
